app/controlplane/pkg/biz/workflowcontract.go

```diff
@@ -346,6 +359,16 @@ func (uc *WorkflowContractUseCase) findPolicyGroup(att *schemav1.PolicyGroupAtta
 			uc.logger.Warnf("policy group '%s' skipped since it's not found or it might use an old schema version", att.GetRef())
 			return nil, nil
 		}
+		if remoteGroup.PolicyGroup != nil {
+			// validate group arguments
+			with := att.GetWith()
+			for _, input := range remoteGroup.PolicyGroup.GetSpec().GetInputs() {
```

Comment:
we should extract this somehow to common logic. It's repeated

Reply:
I'll do it in a separate PR, thanks